src/resolvers.ts

```diff
@@ -69,9 +82,75 @@ const resolvers = {
             const newHealthcareProfessional = await addHealthcareProfessional(args.input)
 
             return newHealthcareProfessional
+        },
+        createSubmission: async (_parent: SubmissionInput, args: {
+            input:{
+                googleMapsUrl: string,
+                healthcareProfessionalName: string,
+                spokenLanguages: SpokenLanguageInput[]
+            }
+        }) => {
+            const submissionData: Submission = {
+                id: '',
+                googleMapsUrl: args.input.googleMapsUrl,
+                healthcareProfessionalName: args.input.healthcareProfessionalName,
+                spokenLanguages: args.input.spokenLanguages
+                    .filter(lang => lang !== null)
+                    .map(lang => ({
+                        iso639_3: lang.iso639_3,
+                        nameJa: lang.nameJa,
+                        nameEn: lang.nameEn,
+                        nameNative: lang.nameNative
+                    })),
+                isUnderReview: true,
+                isApproved: false,
+                isRejected: false
+            }
+
+            const newSubmission = await addSubmission(submissionData)
+
+            return newSubmission
+        },
+        updateSubmission: async (_parent: SubmissionInput, args: {
+            id: string,
+            input: {
+                googleMapsUrl?: string,
+                healthcareProfessionalName?: string,
+                spokenLanguages?: SpokenLanguageInput[],
+                isUnderReview?: boolean,
+                isApproved?: boolean,
+                isRejected?: boolean
+            }
+        }) => {
+            let updatedSpokenLanguages
+
+            if (args.input.spokenLanguages) {
```

Comment:
good thinking here!
you can move this validation logic into the submission service as a `mapAndValidateSpokenLanguages` method and call it from  here. 
Also, consider what happens if it fails, what should we return? likely we'll need to send some sort of error to the user and not run the update